src/vmm/src/builder.rs

```diff
@@ -1087,6 +1098,7 @@ pub mod tests {
             uffd: None,
             vcpus_handles: Vec::new(),
             vcpus_exit_evt,
+            resource_allocator: Rc::new(ResourceAllocator::new().unwrap()),
```

Comment:
Even thought it is just a test, you already break your own requirement for `ResourceAllocator` to be shared between `Vmm` and `MMIODeviceManager`.